Backend/lib/passport.js

```diff
@@ -0,0 +1,53 @@
+const models = require('../models');
+const crypto = require('crypto')
+
+module.exports = (app) => {
+
+  const passport = require('passport');
+  LocalStrategy = require('passport-local').Strategy;
+
+  app.use(passport.initialize());
+  app.use(passport.session());
+
+  passport.serializeUser((user, done) => {
+    // console.log('serialize : ', user);
+    done(null, user.email);
+  });
+
+  passport.deserializeUser((id, done) => {
+    console.log("deserialize", id);
+    models.user.findAll({
+      where: {
+        email: id
+      }
+    })
+      .then(user => done(null, user))
+      .catch(e => done(e, false))
+  });
+
+  passport.use(new LocalStrategy(
+    {
+      usernameField: 'email',
+      passwordField: 'password',
+      passReqToCallback: true
+    },
+    (req, username, password, done) => {
+      let hashPassword = crypto.createHash("sha512").update(req.query.password).digest("hex");
+      models.user.findAll({
+        where: {
+          email: req.query.email
+        }
+      }).then(result => {
+        if (result == "") {
```

Comment:
is there a case where result is null?

Reply:
if DB do not have certain email, it's result is null.

Reply:
Again, when I ask a question, I want you to update the code so I don't have to ask this question. If there is a null case, please take care of it.